### ISSUE TITLE
[feature request]arg_scope in gluon

### ISSUE THREAD

**Issue by dongzhuoyao** [apache/incubator-mxnet#9767]
what if I want to modify the batch normalization behavior in the vision model zoo.

an arg_scope function  in tf.slim  will help a lot.

**Comment by szha** [comment]
@dongzhuoyao I'm not sure if I follow. How do you intend to change the batch-norm behavior? Would you provide an example?

**Comment by piiswrong** [comment]
@szha See https://stackoverflow.com/questions/45226950/what-does-arg-scope-actually-do
I think this would be useful although I'm not sure how it should be implemented

**Comment by szha** [comment]
Seems doable with decorators and scope.
http://code.activestate.com/recipes/577382-keyword-argument-injection-with-python-decorators/

With this solution, we'd need to decorate every constructor in Gluon.nn/rnn though.
